x/evm/keeper/state_transition.go

```diff
@@ -220,13 +220,29 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 	res.Hash = txHash.Hex()
 
 	logs := k.GetTxLogsTransient(txHash)
+	receipt := &ethtypes.Receipt{
+		Type:      tx.Type(),
+		PostState: nil,
+		// NOTE: assume until this point that tx is successful
+		Status: ethtypes.ReceiptStatusSuccessful,
+		Bloom:  ethtypes.BytesToBloom(k.GetBlockBloomTransient().Bytes()),
+		// Cumulative gas used for the current block, excluding this tx
+		CumulativeGasUsed: ctx.BlockGasMeter().GasConsumedToLimit(),
+		TxHash:            txHash,
+		Logs:              logs,
+		GasUsed:           res.GasUsed,
+		ContractAddress:   *tx.To(),
```

Comment:
TODO: fix potential panic

Reply:
we dont know if its a `ContractAddress` at this stage.